onadata/apps/api/tasks.py

```diff
@@ -106,3 +111,21 @@ def send_verification_email(email, message_txt, subject):
 def send_account_lockout_email(email, message_txt, subject):
     """Sends account locked email."""
     send_generic_email(email, message_txt, subject)
+
+
+@app.task()
+def delete_inactive_submissions():
+    """
+    Task to periodically delete soft deleted submissions from db
+    """
+    submissions_lifespan = getattr(settings, "INACTIVE_SUBMISSIONS_LIFESPAN", None)
+    if submissions_lifespan:
+        time_threshold = timezone.now() - timedelta(days=submissions_lifespan)
+        # delete instance attachments
+        instances = Instance.objects.filter(
+            Q(deleted_at__isnull=False) | Q(deleted_at__gte=time_threshold),
+            date_created__lte=time_threshold,
```

Comment:
Should this check be based only on the time since the record was last deleted hence using the `deleted_at` field for checks and not when the record was created?